controller_manager:
  ros__parameters:
    update_rate: 200 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # The JointTrajectoryController allows you to send joint trajectory commands to a group 
    # of joints on a robot. These commands specify the desired positions for each joint.
    arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController


arm_controller:
  ros__parameters:
    joints:
      - joint2_to_joint1
      - joint3_to_joint2
      - joint4_to_joint3
      - joint5_to_joint4
      - joint6_to_joint5
      - joint6output_to_joint6
    
    # The controller will expect position commands as input for each of these joints.
    command_interfaces:
      - position

    # Tells the controller that it should expect to receive position and velocity data as the state 
    # feedback from the hardware interface,
    state_interfaces:
      - position

    state_publish_rate: 100.0 # Defaults to 50
    action_monitor_rate: 100.0 # Defaults to 20

    allow_partial_joints_goal: false # Defaults to false
    hardware_state_has_offset: true
    deduce_states_from_derivatives: true
    
    # If true, When set to true, the controller will not use any feedback from the system 
    # (e.g., joint positions, velocities, efforts) to compute the control commands.
    # open_loop_control: true

    constraints:
      stopped_velocity_tolerance: 0.01 # Defaults to 0.01
      goal_time: 0.0 # Defaults to 0.0 (start immediately)